src/main/kotlin/org/dashevo/dpp/contract/ContractFactory.kt

```diff
@@ -9,49 +9,37 @@ package org.dashevo.dpp.contract
 
 import org.dashevo.dpp.Factory
 import org.dashevo.dpp.util.Cbor
+import org.dashevo.dpp.util.Entropy
+import org.dashevo.dpp.util.HashUtils
 
 class ContractFactory : Factory() {
 
-    fun createDataContract(rawDataContract: MutableMap<String, Any?>): Contract {
+    fun createDataContract(ownerId: String, rawDataContract: MutableMap<String, Any?>): DataContract {
 
-        val contractId = if (rawDataContract.containsKey("contractId")) rawDataContract["contractId"] as String else ""
+        val dataContractEntropy = Entropy.generate()
+        val dataContractId = HashUtils.generateDataContractId(ownerId, dataContractEntropy)
 
-        val contract = Contract(contractId,
+        val dataContract = DataContract(dataContractId,
+                ownerId,
+                DataContract.SCHEMA,
                 rawDataContract["documents"] as MutableMap<String, Any?>)
 
-        if (rawDataContract.containsKey("\$schema")) {
-            contract.schema = rawDataContract["\$schema"] as String
-        }
+        dataContract.entropy = dataContractEntropy
 
-        if (rawDataContract.containsKey("version")) {
-            contract.version = rawDataContract["version"] as Int
-        }
-
-        if (rawDataContract.containsKey("definitions")) {
-            contract.definitions = rawDataContract["definitions"] as MutableMap<String, Any>
-        }
-
-        return contract
-    }
-
-    fun create(contractId: String, documents: MutableMap<String, Any?>): Contract {
-        val rawContract = HashMap<String, Any?>(2)
-        rawContract["contractId"] = contractId
-        rawContract["documents"] = documents
-        return createDataContract(rawContract)
+        return dataContract
     }
 
-    fun createFromObject(rawContract: MutableMap<String, Any?>, options: Options = Options()): Contract {
-        return createDataContract(rawContract)
+    fun createFromObject(rawContract: MutableMap<String, Any?>, options: Options = Options()): DataContract {
+        return DataContract(rawContract)
     }
 
-    fun createFromSerialized(payload: ByteArray, options: Options = Options()): Contract {
+    fun createFromSerialized(payload: ByteArray, options: Options = Options()): DataContract {
         val rawDocument = Cbor.decode(payload).toMutableMap()
         return createFromObject(rawDocument, options)
     }
 
-    fun createStateTransition(contract: Contract) : ContractStateTransition {
-        return ContractStateTransition(contract)
+    fun createStateTransition(dataContract: DataContract) : ContractStateTransition {
```

Comment:
Any reason we're not calling the constructor directly?

Reply:
This is calling the ContactStateTransition constructor.  Contract state transitions were not updated as part of this PR.  We have no tests for it and no other library uses this code.  It probably doesn't work.

This is the associated JavaScript:
```
  /**
   * Create Data Contract State Transition
   *
   * @param {DataContract} dataContract
   * @return {DataContractCreateTransition}
   */
  createStateTransition(dataContract) {
    return new DataContractCreateTransition({
      dataContract: dataContract.toJSON(),
      entropy: dataContract.getEntropy(),
    });
  }
```